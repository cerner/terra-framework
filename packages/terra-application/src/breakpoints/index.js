export { default } from 'terra-breakpoints';
export * from 'terra-breakpoints';
