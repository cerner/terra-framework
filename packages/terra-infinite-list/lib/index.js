"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "Item", {
  enumerable: true,
  get: function get() {
    return _terraList.Item;
  }
});
Object.defineProperty(exports, "SectionHeader", {
  enumerable: true,
  get: function get() {
    return _terraList.SectionHeader;
  }
});
Object.defineProperty(exports, "SubsectionHeader", {
  enumerable: true,
  get: function get() {
    return _terraList.SubsectionHeader;
  }
});
exports.default = void 0;
var _terraList = require("terra-list");
var _InfiniteList = _interopRequireDefault(require("./InfiniteList"));
var _default = _InfiniteList.default;
exports.default = _default;