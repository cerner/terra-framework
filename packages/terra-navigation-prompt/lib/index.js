"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "NavigationPromptCheckpoint", {
  enumerable: true,
  get: function get() {
    return _NavigationPromptCheckpoint.default;
  }
});
Object.defineProperty(exports, "PromptRegistrationContext", {
  enumerable: true,
  get: function get() {
    return _PromptRegistrationContext.default;
  }
});
exports.default = void 0;
var _NavigationPrompt = _interopRequireDefault(require("./NavigationPrompt"));
var _NavigationPromptCheckpoint = _interopRequireDefault(require("./NavigationPromptCheckpoint"));
var _PromptRegistrationContext = _interopRequireDefault(require("./PromptRegistrationContext"));
var _default = _NavigationPrompt.default;
exports.default = _default;