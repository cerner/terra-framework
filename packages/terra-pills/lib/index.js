"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "Pill", {
  enumerable: true,
  get: function get() {
    return _Pill.default;
  }
});
exports.default = void 0;
var _FilterPills = _interopRequireDefault(require("./FilterPills"));
var _Pill = _interopRequireDefault(require("./subcomponents/_Pill"));
var _default = _FilterPills.default;
exports.default = _default;