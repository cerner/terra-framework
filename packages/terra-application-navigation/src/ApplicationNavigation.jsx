import React, {
  useEffect, useLayoutEffect, useState, useRef,
} from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import Overlay from 'terra-overlay';
import { ActiveBreakpointContext } from 'terra-application/lib/breakpoints';
import FocusTrap from 'focus-trap-react';
import Popup from 'terra-popup';
import VisuallyHiddenText from 'terra-visually-hidden-text';

import Header from './header/_Header';
import CompactHeader from './header/_CompactHeader';
import DrawerMenu from './drawer-menu/_DrawerMenu';
import UtilityMenu from './utility-menu/_UtilityMenu';
import { shouldRenderCompactNavigation } from './utils/helpers';
import {
  titleConfigPropType, userConfigPropType, navigationItemsPropType, extensionItemsPropType, utilityItemsPropType,
} from './utils/propTypes';

import styles from './ApplicationNavigation.module.scss';

const cx = classNames.bind(styles);

const propTypes = {
  /**
   * Key/Value pairs associating a string key entry to a numerical notification count.
   */
  notifications: PropTypes.object,
  /**
   * A configuration object with information pertaining to the application's title.
   */
  titleConfig: titleConfigPropType,
  /**
   * A configuration object with information specifying the creation of the Extension buttons rendered within the
   * ApplicationNavigation header.
   */
  extensionItems: extensionItemsPropType,
  /**
   * A function to be executed upon the selection of an extensions item.
   * Ex: `onSelectExtensionsItem(String selectedUtilityItemKey, Object metaData)`
   */
  onSelectExtensionItem: PropTypes.func,
  /**
   * A configuration object with information pertaining to the application's user.
   */
  userConfig: userConfigPropType,
  /**
   * An element to render within the ApplicationNavigation menu, shifted to the drawer at the `medium` and below.
   */
  hero: PropTypes.element,
  /**
   * An array of configuration objects with information specifying the creation of navigation items. These items
   * are rendered within the ApplicationNavigation header at larger breakpoints and within the drawer menu at smaller breakpoints.
   */
  navigationItems: navigationItemsPropType,
  /**
   * A function to be executed for the render of each navigation item.
   */
  navigationRenderFunction: PropTypes.func,
  /**
   * A string key representing the currently active navigation item. This value should match one of the item keys provided in the
   * `navigationItems` array.
   */
  activeNavigationItemKey: PropTypes.string,
  /**
  * Callback function triggered on Drawer Menu state change
  */
  onDrawerMenuStateChange: PropTypes.func,
  /**
   * A function to be executed upon the selection of a navigation item.
   * Ex: `onSelectNavigationItem(String selectedNavigationItemKey, Object metaData)`
   */
  onSelectNavigationItem: PropTypes.func,
  /**
   * A function to be executed upon the selection of the Settings utility item.
   * If `onSelectSettings` is not provided, the Settings utility item will not be rendered.
   * Ex: `onSelectSettings()`
   */
  onSelectSettings: PropTypes.func,
  /**
   * A function to be executed upon the selection of the Help utility item.
   * If `onSelectHelp` is not provided, the Help utility item will not be rendered.
   * Ex: `onSelectLogout()`
   */
  onSelectHelp: PropTypes.func,
  /**
   * A function to be executed upon the selection of the Logout action button.
   * If `onSelectLogout` is not provided, the Logout action button will not be rendered.
   * Ex: `onSelectLogout()`
   */
  onSelectLogout: PropTypes.func,
  /**
   * An array of configuration objects with information specifying the creation of additional utility menu items.
   * These items are rendered within the popup utility menu at larger breakpoints and within the drawer menu at smaller breakpoints.
   */
  utilityItems: utilityItemsPropType,
  /**
   * A function to be executed upon the selection of a custom utility item.
   * Ex: `onSelectUtilityItem(String selectedUtilityItemKey, Object metaData)`
   */
  onSelectUtilityItem: PropTypes.func,
  /**
   * A collection of child elements to render within the ApplicationNavigation body.
   */
  children: PropTypes.node,
};

const defaultProps = {
  utilityItems: [],
  notifications: {},
};

const ApplicationNavigation = ({
  titleConfig,
  userConfig,
  hero,
  navigationItems,
  onSelectNavigationItem,
  activeNavigationItemKey,
  navigationRenderFunction,
  extensionItems,
  onDrawerMenuStateChange,
  onSelectExtensionItem,
  onSelectSettings,
  onSelectHelp,
  onSelectLogout,
  utilityItems,
  onSelectUtilityItem,
  notifications,
  children,
}) => {
  const drawerMenuRef = useRef();
  const contentLayoutRef = useRef();
  const mainContainerRef = useRef();
  const utilityButtonPopupAnchorRef = useRef();
  // Need to capture the animation life-cycle as opposed to the open/close state.
  const drawerMenuIsVisibleRef = useRef(false);
  // FocusTrap captures the initial value of the onDeactivate callback, so need a persistent ref to the isOpen value.
  const drawerMenuIsOpenRef = useRef(false);
  const closeMenuCallbackRef = useRef();

  const [drawerMenuIsOpen, setDrawerMenuIsOpen] = useState(false);
  const [popupMenuIsOpen, setPopupMenuIsOpen] = useState(false);

  // Use dot notation temporarily until hooks + enzyme support for userContext
  const activeBreakpoint = React.useContext(ActiveBreakpointContext);

  function updateDrawerIsOpen(value) {
    drawerMenuIsOpenRef.current = value;
    setDrawerMenuIsOpen(value);
    if (onDrawerMenuStateChange) {
      onDrawerMenuStateChange(value);
    }
  }

  /**
   * Given a callback function, generateMenuClosingCallback will return a new function
   * that will ensure that the various menu states are reset before the callback function
   * is executed.
   */
  function generateMenuClosingCallback(wrappedFunction) {
    return (...args) => {
      if (!wrappedFunction) {
        return;
      }

      closeMenuCallbackRef.current = () => { wrappedFunction(...args); };
      updateDrawerIsOpen(false);
      setPopupMenuIsOpen(false);
    };
  }

  function focusMainContent() {
    if (mainContainerRef.current) {
      mainContainerRef.current.focus();
    }
  }

  function focusToggle() {
    const toggle = document.querySelector('[data-compact-header-toggle="true"]');
    if (toggle) {
      toggle.focus();
    }
  }

  function generateFocusToggle(wrappedFunction) {
    return (...args) => {
      wrappedFunction(...args);
      window.requestAnimationFrame(() => {
        focusToggle();
      });
    };
  }

  function generateFocusMain(wrappedFunction) {
    return (...args) => {
      wrappedFunction(...args);
      window.requestAnimationFrame(() => {
        focusMainContent();
      });
    };
  }

  function renderDrawerMenu() {
    if (!shouldRenderCompactNavigation(activeBreakpoint)) {
      return null;
    }

    return (
      <FocusTrap
        active={drawerMenuIsOpen}
        focusTrapOptions={{
          escapeDeactivates: true,
          clickOutsideDeactivates: true,
          returnFocusOnDeactivate: false,
        }}
        className={cx('drawer-menu-focus-trap-container')}
      >
        <DrawerMenu
          titleConfig={titleConfig}
          userConfig={userConfig}
          hero={hero}
          navigationItems={navigationItems}
          activeNavigationItemKey={activeNavigationItemKey}
          onSelectNavigationItem={onSelectNavigationItem ? generateFocusMain(generateMenuClosingCallback(onSelectNavigationItem)) : undefined}
          onSelectSettings={onSelectSettings ? generateMenuClosingCallback(onSelectSettings) : undefined}
          onSelectHelp={onSelectHelp ? generateMenuClosingCallback(onSelectHelp) : undefined}
          onSelectLogout={onSelectLogout ? generateMenuClosingCallback(onSelectLogout) : undefined}
          utilityItems={utilityItems}
          onSelectUtilityItem={onSelectUtilityItem ? generateMenuClosingCallback(onSelectUtilityItem) : undefined}
          notifications={notifications}
        />
      </FocusTrap>
    );
  }

  function renderPopupMenu() {
    return (
      <Popup
        attachmentBehavior="push"
        contentAttachment="top right"
        contentHeight="auto"
        contentWidth="320"
        isArrowDisplayed
        isHeaderDisabled
        isOpen={popupMenuIsOpen}
        onRequestClose={() => setPopupMenuIsOpen(false)}
        targetAttachment="bottom center"
        targetRef={() => utilityButtonPopupAnchorRef.current}
        isContentFocusDisabled
      >
        <UtilityMenu
          hero={hero}
          userConfig={userConfig}
          onSelectSettings={onSelectSettings ? generateMenuClosingCallback(onSelectSettings) : undefined}
          onSelectHelp={onSelectHelp ? generateMenuClosingCallback(onSelectHelp) : undefined}
          onSelectLogout={onSelectLogout ? generateMenuClosingCallback(onSelectLogout) : undefined}
          utilityItems={utilityItems}
          onSelectUtilityItem={onSelectUtilityItem ? generateMenuClosingCallback(onSelectUtilityItem) : undefined}
        />
      </Popup>
    );
  }

  function renderCompactHeader() {
    return (
      <CompactHeader
        activeBreakpoint={activeBreakpoint}
        titleConfig={titleConfig}
        extensionItems={extensionItems}
        onSelectExtensionItem={onSelectExtensionItem}
        navigationItems={navigationItems}
        onSelectMenuButton={() => updateDrawerIsOpen(true)}
        onSelectSkipToContent={focusMainContent}
        notifications={notifications}
        isDrawerMenuOpen={drawerMenuIsOpen}
        utilityItems={utilityItems}
        activeNavigationItemKey={activeNavigationItemKey}
        userConfig={userConfig}
        hero={hero}
        onSelectNavigationItem={generateFocusMain(onSelectNavigationItem)}
        onSelectUtilityItem={onSelectUtilityItem}
        onSelectSettings={onSelectSettings}
        onSelectHelp={onSelectHelp}
        onSelectLogout={onSelectLogout}
      />
    );
  }

  function renderHeader() {
    return (
      <Header
        activeBreakpoint={activeBreakpoint}
        titleConfig={titleConfig}
        extensionItems={extensionItems}
        onSelectExtensionItem={onSelectExtensionItem}
        navigationItems={navigationItems}
        navigationRenderFunction={navigationRenderFunction}
        activeNavigationItemKey={activeNavigationItemKey}
        onSelectNavigationItem={generateFocusMain(onSelectNavigationItem)}
        userConfig={userConfig}
        onSelectSkipToContent={focusMainContent}
        notifications={notifications}
        utilityButtonPopupAnchorRef={utilityButtonPopupAnchorRef}
        onSelectUtilityButton={() => setPopupMenuIsOpen(true)}
        hero={hero}
        utilityItems={utilityItems}
        onSelectSettings={onSelectSettings}
        onSelectHelp={onSelectHelp}
        onSelectLogout={onSelectLogout}
      />
    );
  }

  /**
   * This effect is used to execute callbacks from the drawer and popup
   * menus after they are closed. This is similar to executing the callbacks
   * in a setState callback for the menu state, but setState callbacks do not
   * exist for the useState hook.
   *
   * The closeMenuCallbackRef value is set by the functions returned by
   * generateMenuClosingCallback.
   */
  useEffect(() => {
    if (closeMenuCallbackRef.current) {
      closeMenuCallbackRef.current();
      closeMenuCallbackRef.current = undefined;
    }
  });

  /**
   * This layout effect is used to manage the visibility of the drawer menu during
   * its transition animations.
   *
   * When the drawer menu is closed, it is hidden using its display property to
   * prevent users from tabbing to it and to limit expensive reflows of its content. However, while
   * the drawerMenuIsOpen state indicates whether or not the menu is open or closed, the transitions
   * to the open/closed states are animated, leaving the component in a position where the menu must
   * continue to be visible while the menu is animating closed. Therefore, we cannot rely on the
   * drawerMenuIsOpen state alone to determine when to hide the drawer menu content.
   *
   * In this layout effect, the proper styles are applied after the transition completes, and the
   * drawerMenuIsVisibleRef value is updated so that subsequent renders of the component are
   * accurate.
   */
  useLayoutEffect(() => {
    const contentLayoutElement = contentLayoutRef.current;

    if (!contentLayoutElement) {
      return undefined;
    }

    function cleanupContentTransition(event) {
      if (event.currentTarget.classList.contains(cx('drawer-menu-is-open'))) {
        drawerMenuRef.current.removeAttribute('style');
        drawerMenuIsVisibleRef.current = true;
      } else {
        drawerMenuRef.current.style.display = 'none';
        drawerMenuIsVisibleRef.current = false;
      }
    }

    contentLayoutElement.addEventListener('transitionend', cleanupContentTransition);

    return () => {
      contentLayoutElement.removeEventListener('transitionend', cleanupContentTransition);
    };
  }, [contentLayoutRef]);

  /**
   * If the ApplicationNavigation is rendering at non-compact breakpoints, and the drawer menu is still
   * open, then it is closed here. This will trigger an new update immediately after the completion of
   * the current render.
   */
  if (drawerMenuIsOpen && !shouldRenderCompactNavigation(activeBreakpoint)) {
    updateDrawerIsOpen(false);
  }

  /**
   * The inline display styles are removed from the drawerMenuRef with each render. This
   * ensures that the menu is immediately visible if it is being opened. If the menu still
   * needs to be hidden, the logic below will provide the appropriate display style.
   */
  if (drawerMenuRef.current && drawerMenuIsOpen) {
    drawerMenuRef.current.removeAttribute('style');
  }

  let hiddenMainTitle;
  if (navigationItems && activeNavigationItemKey) {
    hiddenMainTitle = (
      <VisuallyHiddenText
        id="main-inner-title"
        aria-hidden
        text={navigationItems.filter((item => item.key === activeNavigationItemKey))[0].text}
      />
    );
  }

  const handleRequestClose = (event) => {
    event.preventDefault();
    event.stopPropagation();
    updateDrawerIsOpen(false);
  };

  return (
    <div className={cx('application-navigation')}>
      <div
        ref={drawerMenuRef}
        className={cx('drawer-menu-container')}
        style={!drawerMenuIsVisibleRef.current && !drawerMenuIsOpen ? { display: 'none' } : undefined} // eslint-disable-line react/forbid-dom-props
        aria-hidden={!drawerMenuIsOpen ? true : null}
      >
        {renderDrawerMenu()}
      </div>
      <div
        ref={contentLayoutRef}
        className={cx('content-layout', { 'drawer-menu-is-open': drawerMenuIsOpen })}
      >
        {shouldRenderCompactNavigation(activeBreakpoint) ? renderCompactHeader() : renderHeader()}
        <main
          ref={mainContainerRef}
          tabIndex="-1"
          role="main"
          className={cx('main-container')}
          aria-labelledby={hiddenMainTitle ? 'main-inner-title' : null}
        >
          {hiddenMainTitle}
          {children}
        </main>
        <Overlay
          className={cx('overlay')}
          isOpen={drawerMenuIsOpen}
          isRelativeToContainer
          backgroundStyle="clear"
          zIndex="6000"
          onRequestClose={generateFocusToggle(handleRequestClose)}
        />
      </div>
      {renderPopupMenu()}
    </div>
  );
};

ApplicationNavigation.propTypes = propTypes;
ApplicationNavigation.defaultProps = defaultProps;

/**
 * The <main> tag rendered by ApplicationNavigation is not officially supported by legacy Internet Explorer browsers.
 * React will log warnings if we attempt to render it. However, by creating a main element here prior to rendering the component,
 * Internet Explorer is tricked into recognizing the <main> tag, and React does not warn.
 */
document.createElement('main');

export default ApplicationNavigation;
