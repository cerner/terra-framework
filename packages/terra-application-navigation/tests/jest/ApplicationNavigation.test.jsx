import React from 'react';
import { ActiveBreakpointContext } from 'terra-application/lib/breakpoints';
import ApplicationNavigation from '../../src/ApplicationNavigation';

describe('ApplicationNavigation', () => {
  let reactUseContext;
  let breakpointContextValue;

  beforeAll(() => {
    /**
     * Until Enzyme is updated with full support for hooks, we need to
     * mock out the useContext implementation.
     */
    reactUseContext = React.useContext;
    React.useContext = (contextValue) => {
      if (ActiveBreakpointContext === contextValue) {
        return breakpointContextValue;
      }
      return reactUseContext(contextValue);
    };
  });

  afterAll(() => {
    React.useContext = reactUseContext;
  });

  it('should render default element', () => {
    const shallowComponent = shallow(
      <ApplicationNavigation />,
    );

    expect(shallowComponent).toMatchSnapshot();
  });

  it('should render with prop data at large breakpoint', () => {
    breakpointContextValue = 'large';

    const shallowComponent = shallow(
      <ApplicationNavigation
        activeBreakpoint="large"
        titleConfig={{
          title: 'test-title',
        }}
        activeNavigationItemKey="my-test-key"
        hero={<div>my test hero</div>}
        userConfig={{
          name: 'user-name',
          detail: 'user-detail',
          initials: 'user-initials',
          imageSrc: 'user-src',
        }}
        navigationItems={[{ text: 'test-text', key: 'my-test-key' }]}
        extensionItems={[{ text: 'test-text-1', key: 'my-test-key-1', icon: <span>my icon</span> }]}
        utilityItems={[{ text: 'test-text-2', key: 'my-test-key-2', icon: <span>my icon</span> }]}
        notifications={{ 'my-test-key-1': 3 }}
        navigationRenderFunction={jest.fn()}
        onSelectNavigationItem={jest.fn()}
        onSelectExtensionItem={jest.fn()}
        onSelectUtilityItem={jest.fn()}
        onSelectSettings={jest.fn()}
        onSelectHelp={jest.fn()}
        onSelectLogout={jest.fn()}
      >
        <div>test child content</div>
      </ApplicationNavigation>,
    );

    expect(shallowComponent).toMatchSnapshot();
  });

  it('should render with prop data at medium breakpoint', () => {
    breakpointContextValue = 'medium';

    const shallowComponent = shallow((
      <ApplicationNavigation
        titleConfig={{
          title: 'test-title',
        }}
        activeNavigationItemKey="my-test-key"
        hero={<div>my test hero</div>}
        userConfig={{
          name: 'user-name',
          detail: 'user-detail',
          initials: 'user-initials',
          imageSrc: 'user-src',
        }}
        navigationItems={[{ text: 'test-text', key: 'my-test-key' }]}
        extensionItems={[{ text: 'test-text-1', key: 'my-test-key-1', icon: <span>my icon</span> }]}
        utilityItems={[{ text: 'test-text-2', key: 'my-test-key-2', icon: <span>my icon</span> }]}
        notifications={{ 'my-test-key-1': 3 }}
        navigationRenderFunction={jest.fn()}
        onSelectNavigationItem={jest.fn()}
        onSelectExtensionItem={jest.fn()}
        onSelectUtilityItem={jest.fn()}
        onSelectSettings={jest.fn()}
        onSelectHelp={jest.fn()}
        onSelectLogout={jest.fn()}
      >
        <div>test child content</div>
      </ApplicationNavigation>
    ));

    expect(shallowComponent).toMatchSnapshot();
  });
});
