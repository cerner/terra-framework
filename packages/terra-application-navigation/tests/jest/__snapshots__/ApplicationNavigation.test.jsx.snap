// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationNavigation should render default element 1`] = `
<div
  className="application-navigation"
>
  <div
    aria-hidden={true}
    className="drawer-menu-container"
    style={
      Object {
        "display": "none",
      }
    }
  />
  <div
    className="content-layout"
  >
    <InjectIntl(Header)
      notifications={Object {}}
      onSelectNavigationItem={[Function]}
      onSelectSkipToContent={[Function]}
      onSelectUtilityButton={[Function]}
      utilityButtonPopupAnchorRef={
        Object {
          "current": undefined,
        }
      }
      utilityItems={Array []}
    />
    <main
      aria-labelledby={null}
      className="main-container"
      role="main"
      tabIndex="-1"
    />
    <Overlay
      backgroundStyle="clear"
      className="overlay"
      isOpen={false}
      isRelativeToContainer={true}
      isScrollable={false}
      onRequestClose={[Function]}
      rootSelector="#root"
      zIndex="6000"
    />
  </div>
  <Popup
    attachmentBehavior="push"
    boundingRef={null}
    classNameArrow={null}
    classNameContent={null}
    classNameOverlay={null}
    contentAttachment="top right"
    contentHeight="auto"
    contentWidth="320"
    isArrowDisplayed={true}
    isContentFocusDisabled={true}
    isHeaderDisabled={true}
    isOpen={false}
    onRequestClose={[Function]}
    popupContentRole="dialog"
    targetAttachment="bottom center"
    targetRef={[Function]}
  >
    <InjectIntl(UtilityMenu)
      utilityItems={Array []}
    />
  </Popup>
</div>
`;

exports[`ApplicationNavigation should render with prop data at large breakpoint 1`] = `
<div
  className="application-navigation"
>
  <div
    aria-hidden={true}
    className="drawer-menu-container"
    style={
      Object {
        "display": "none",
      }
    }
  />
  <div
    className="content-layout"
  >
    <InjectIntl(Header)
      activeBreakpoint="large"
      activeNavigationItemKey="my-test-key"
      extensionItems={
        Array [
          Object {
            "icon": <span>
              my icon
            </span>,
            "key": "my-test-key-1",
            "text": "test-text-1",
          },
        ]
      }
      hero={
        <div>
          my test hero
        </div>
      }
      navigationItems={
        Array [
          Object {
            "key": "my-test-key",
            "text": "test-text",
          },
        ]
      }
      navigationRenderFunction={[MockFunction]}
      notifications={
        Object {
          "my-test-key-1": 3,
        }
      }
      onSelectExtensionItem={[MockFunction]}
      onSelectHelp={[MockFunction]}
      onSelectLogout={[MockFunction]}
      onSelectNavigationItem={[Function]}
      onSelectSettings={[MockFunction]}
      onSelectSkipToContent={[Function]}
      onSelectUtilityButton={[Function]}
      titleConfig={
        Object {
          "title": "test-title",
        }
      }
      userConfig={
        Object {
          "detail": "user-detail",
          "imageSrc": "user-src",
          "initials": "user-initials",
          "name": "user-name",
        }
      }
      utilityButtonPopupAnchorRef={
        Object {
          "current": undefined,
        }
      }
      utilityItems={
        Array [
          Object {
            "icon": <span>
              my icon
            </span>,
            "key": "my-test-key-2",
            "text": "test-text-2",
          },
        ]
      }
    />
    <main
      aria-labelledby="main-inner-title"
      className="main-container"
      role="main"
      tabIndex="-1"
    >
      <VisuallyHiddenText
        aria-hidden={true}
        id="main-inner-title"
        text="test-text"
      />
      <div>
        test child content
      </div>
    </main>
    <Overlay
      backgroundStyle="clear"
      className="overlay"
      isOpen={false}
      isRelativeToContainer={true}
      isScrollable={false}
      onRequestClose={[Function]}
      rootSelector="#root"
      zIndex="6000"
    />
  </div>
  <Popup
    attachmentBehavior="push"
    boundingRef={null}
    classNameArrow={null}
    classNameContent={null}
    classNameOverlay={null}
    contentAttachment="top right"
    contentHeight="auto"
    contentWidth="320"
    isArrowDisplayed={true}
    isContentFocusDisabled={true}
    isHeaderDisabled={true}
    isOpen={false}
    onRequestClose={[Function]}
    popupContentRole="dialog"
    targetAttachment="bottom center"
    targetRef={[Function]}
  >
    <InjectIntl(UtilityMenu)
      hero={
        <div>
          my test hero
        </div>
      }
      onSelectHelp={[Function]}
      onSelectLogout={[Function]}
      onSelectSettings={[Function]}
      onSelectUtilityItem={[Function]}
      userConfig={
        Object {
          "detail": "user-detail",
          "imageSrc": "user-src",
          "initials": "user-initials",
          "name": "user-name",
        }
      }
      utilityItems={
        Array [
          Object {
            "icon": <span>
              my icon
            </span>,
            "key": "my-test-key-2",
            "text": "test-text-2",
          },
        ]
      }
    />
  </Popup>
</div>
`;

exports[`ApplicationNavigation should render with prop data at medium breakpoint 1`] = `
<div
  className="application-navigation"
>
  <div
    aria-hidden={true}
    className="drawer-menu-container"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <FocusTrap
      _createFocusTrap={[Function]}
      active={false}
      className="drawer-menu-focus-trap-container"
      focusTrapOptions={
        Object {
          "clickOutsideDeactivates": true,
          "escapeDeactivates": true,
          "returnFocusOnDeactivate": false,
        }
      }
      paused={false}
    >
      <InjectIntl(DrawerMenu)
        activeNavigationItemKey="my-test-key"
        hero={
          <div>
            my test hero
          </div>
        }
        navigationItems={
          Array [
            Object {
              "key": "my-test-key",
              "text": "test-text",
            },
          ]
        }
        notifications={
          Object {
            "my-test-key-1": 3,
          }
        }
        onSelectHelp={[Function]}
        onSelectLogout={[Function]}
        onSelectNavigationItem={[Function]}
        onSelectSettings={[Function]}
        onSelectUtilityItem={[Function]}
        titleConfig={
          Object {
            "title": "test-title",
          }
        }
        userConfig={
          Object {
            "detail": "user-detail",
            "imageSrc": "user-src",
            "initials": "user-initials",
            "name": "user-name",
          }
        }
        utilityItems={
          Array [
            Object {
              "icon": <span>
                my icon
              </span>,
              "key": "my-test-key-2",
              "text": "test-text-2",
            },
          ]
        }
      />
    </FocusTrap>
  </div>
  <div
    className="content-layout"
  >
    <InjectIntl(CompactHeader)
      activeBreakpoint="medium"
      activeNavigationItemKey="my-test-key"
      extensionItems={
        Array [
          Object {
            "icon": <span>
              my icon
            </span>,
            "key": "my-test-key-1",
            "text": "test-text-1",
          },
        ]
      }
      hero={
        <div>
          my test hero
        </div>
      }
      isDrawerMenuOpen={false}
      navigationItems={
        Array [
          Object {
            "key": "my-test-key",
            "text": "test-text",
          },
        ]
      }
      notifications={
        Object {
          "my-test-key-1": 3,
        }
      }
      onSelectExtensionItem={[MockFunction]}
      onSelectHelp={[MockFunction]}
      onSelectLogout={[MockFunction]}
      onSelectMenuButton={[Function]}
      onSelectNavigationItem={[Function]}
      onSelectSettings={[MockFunction]}
      onSelectSkipToContent={[Function]}
      onSelectUtilityItem={[MockFunction]}
      titleConfig={
        Object {
          "title": "test-title",
        }
      }
      userConfig={
        Object {
          "detail": "user-detail",
          "imageSrc": "user-src",
          "initials": "user-initials",
          "name": "user-name",
        }
      }
      utilityItems={
        Array [
          Object {
            "icon": <span>
              my icon
            </span>,
            "key": "my-test-key-2",
            "text": "test-text-2",
          },
        ]
      }
    />
    <main
      aria-labelledby="main-inner-title"
      className="main-container"
      role="main"
      tabIndex="-1"
    >
      <VisuallyHiddenText
        aria-hidden={true}
        id="main-inner-title"
        text="test-text"
      />
      <div>
        test child content
      </div>
    </main>
    <Overlay
      backgroundStyle="clear"
      className="overlay"
      isOpen={false}
      isRelativeToContainer={true}
      isScrollable={false}
      onRequestClose={[Function]}
      rootSelector="#root"
      zIndex="6000"
    />
  </div>
  <Popup
    attachmentBehavior="push"
    boundingRef={null}
    classNameArrow={null}
    classNameContent={null}
    classNameOverlay={null}
    contentAttachment="top right"
    contentHeight="auto"
    contentWidth="320"
    isArrowDisplayed={true}
    isContentFocusDisabled={true}
    isHeaderDisabled={true}
    isOpen={false}
    onRequestClose={[Function]}
    popupContentRole="dialog"
    targetAttachment="bottom center"
    targetRef={[Function]}
  >
    <InjectIntl(UtilityMenu)
      hero={
        <div>
          my test hero
        </div>
      }
      onSelectHelp={[Function]}
      onSelectLogout={[Function]}
      onSelectSettings={[Function]}
      onSelectUtilityItem={[Function]}
      userConfig={
        Object {
          "detail": "user-detail",
          "imageSrc": "user-src",
          "initials": "user-initials",
          "name": "user-name",
        }
      }
      utilityItems={
        Array [
          Object {
            "icon": <span>
              my icon
            </span>,
            "key": "my-test-key-2",
            "text": "test-text-2",
          },
        ]
      }
    />
  </Popup>
</div>
`;
